spec/compiler/semantic/module_spec.cr

```diff
@@ -1355,18 +1355,122 @@ describe "Semantic: module" do
   it "errors when extending module that defines instance vars (#4065)" do
     assert_error %(
       module Foo
-        @foo : Int32?
+        @x = 0
+      end
 
-        def foo
-          @foo
-        end
+      module Bar
+        extend Foo
       end
+      ),
+      "can't declare instance variables in Foo because it is extended by Bar"
```

Comment:
Active voice generally sounds better. Should be semantically okay.

```suggestion
      "can't declare instance variables in Foo because Bar extends it"
```